pkg/api/login_oauth.go

```diff
@@ -31,7 +31,10 @@ var (
 
 func GenStateString() string {
 	rnd := make([]byte, 32)
-	rand.Read(rnd)
+	if _, err := rand.Read(rnd); err != nil {
+		oauthLogger.Error("failed to generate state string", "err", err)
```

Comment:
I think this function should return the error and fail oauth flow if an error is returned from this function.

Reply:
@marefr I think I fixed this now, please have a look.